actions/check_branch_and_release_type/action.yaml

```diff
@@ -0,0 +1,23 @@
+name: Checks the branch and release type
+description: 'Enforces that pre-release builds are done on a release branch and release builds are done on the default branc'
+inputs:
+  release_type:
+    description: The type of version number to return. Must be one of [Snapshot, Patch, Minor or Major]
+    required: true
+runs:
+  using: "composite" 
+
+  steps:
+    - name: Fail if Pre-release on Default branch
+      if: ${{ inputs.release_type == 'Snapshot' && github.event.repository.default_branch == github.ref_name }}
```

Comment:
Curious why this is not allowed?

Reply:
We discussed it, and we can't think of a case where you would need to create a snapshot from main. You should use a branch for that testing. Code should only be on main that is ready for production.